lib/charms/kratos_external_idp_integrator/v0/kratos_external_provider.py

```diff
@@ -730,25 +731,25 @@ def __init__(self, charm: CharmBase, relation_name: str = DEFAULT_RELATION_NAME)
         self.framework.observe(
             events.relation_changed, self._on_provider_endpoint_relation_changed
         )
-        self.framework.observe(
-            events.relation_departed, self._on_provider_endpoint_relation_changed
-        )
+        self.framework.observe(events.relation_broken, self._on_provider_endpoint_relation_broken)
 
     def _on_provider_endpoint_relation_changed(self, event: RelationEvent) -> None:
         if not event.app:
             return
 
         data = event.relation.data[event.app]
         data = _load_data(data, PROVIDER_JSON_SCHEMA)
-        providers = data["providers"]
 
-        if len(providers) == 0:
+        if not (providers := data["providers"]):
             self.on.client_config_removed.emit(event.relation.id)
             return
 
         p = self._get_provider(providers[0], event.relation)
         self.on.client_config_changed.emit(p)
 
+    def _on_provider_endpoint_relation_broken(self, event: RelationBrokenEvent) -> None:
+        self.on.client_config_removed.emit(event.relation.id)
```

Comment:
suggestion: not sure if you have tested this when kratos is scaled up, but maybe it needs this:
```suggestion
        if event.departing_unit == self.unit:
            self.on.client_config_removed.emit(event.relation.id)
```

Reply:
Yeah, I tested it with scaled-up kratos.

Did not understand the suggested changes. The `event` does not have the attribute `departing_unit`, and the class itself does not have `unit` attribute neither.